html/semantics/forms/form-submission-0/form-double-submit-multiple-targets.html is failing in WebKit

cdumez opened the pull request:
#### 28180db1730cdc22b09a2315e17c55dbe37c4e7c
<pre>
html/semantics/forms/form-submission-0/form-double-submit-multiple-targets.html is failing in WebKit
<a href="https://bugs.webkit.org/show_bug.cgi?id=243471">https://bugs.webkit.org/show_bug.cgi?id=243471</a>

Reviewed by Darin Adler.

Align with Gecko and Blink and allow a single form to submit to multiple frames concurrently
when using form.submit().

Previously, WebKit would always cancel the form&apos;s planned navigation when a new form submission
occurs. However, in other browsers, it seems to depend on the method used to submit the form.
In particular, form.submit() doesn&apos;t cancel the planned navigation while other methods (like
form.requestSubmit() do).

html/semantics/forms/form-submission-0/form-double-submit-multiple-targets.html covers the
case where we should not cancel the planned navigation.

imported/w3c/web-platform-tests/html/semantics/forms/form-submission-0/form-double-submit-multiple-targets.html
covers a case where we&apos;re meant to cancel the planned navigation.

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/forms/form-submission-0/form-double-submit-multiple-targets-expected.txt:
* Source/WebCore/html/HTMLFormElement.cpp:
(WebCore::HTMLFormElement::submitIfPossible):
(WebCore::HTMLFormElement::submit):

Canonical link: <a href="https://commits.webkit.org/253072@main">https://commits.webkit.org/253072@main</a>
</pre>

rniwa reviewed (commented):
r=me

darinadler reviewed (approved):
This seems like a subtle change. Is there enough test coverage? For example, in some cases we will have multiple submissions, and only the last one is pointed to by `m_plannedFormSubmission`. Is that the correct semantic? Or should we be storing all the planned form submissions and canceling them all in `submitIfPossible`? Seems like we need sufficient tests to cover this, not sure if we have them.

cdumez committed:
html/semantics/forms/form-submission-0/form-double-submit-multiple-targets.html is failing in WebKit
https://bugs.webkit.org/show_bug.cgi?id=243471

Reviewed by Darin Adler.

Align with Gecko and Blink and allow a single form to submit to multiple frames concurrently
when using form.submit().

Previously, WebKit would always cancel the form's planned navigation when a new form submission
occurs. However, in other browsers, it seems to depend on the method used to submit the form.
In particular, form.submit() doesn't cancel the planned navigation while other methods (like
form.requestSubmit() do).

html/semantics/forms/form-submission-0/form-double-submit-multiple-targets.html covers the
case where we should not cancel the planned navigation.

imported/w3c/web-platform-tests/html/semantics/forms/form-submission-0/form-double-submit-multiple-targets.html
covers a case where we're meant to cancel the planned navigation.

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/forms/form-submission-0/form-double-submit-multiple-targets-expected.txt:
* Source/WebCore/html/HTMLFormElement.cpp:
(WebCore::HTMLFormElement::submitIfPossible):
(WebCore::HTMLFormElement::submit):

Canonical link: https://commits.webkit.org/253072@main
